From: Keith Busch <keith.busch@intel.com>
Date: Thu, 31 Jan 2013 14:40:38 -0700
Subject: NVMe: Add namespaces with no LBA range feature
References: bnc#913030,FATE#317455
Patch-Mainline: v3.10
Git-commit: 122090366d1d5c6ec1bfb6dfdb3a6d121ff074aa

The LBA Range Type feature is optional in the NVMe specification,
so we should continue with adding namespaces for controllers that do
not implement this feature.

Signed-off-by: Keith Busch <keith.busch@intel.com>
Signed-off-by: Matthew Wilcox <matthew.r.wilcox@intel.com>
Acked-by: Hannes Reinecke <hare@suse.de>
---
 drivers/block/nvme.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/drivers/block/nvme.c b/drivers/block/nvme.c
index be4f20a..7879e81 100644
--- a/drivers/block/nvme.c
+++ b/drivers/block/nvme.c
@@ -1546,7 +1546,7 @@ static int __devinit nvme_dev_add(struct nvme_dev *dev)
 		res = nvme_get_features(dev, NVME_FEAT_LBA_RANGE, i,
 							dma_addr + 4096, NULL);
 		if (res)
-			continue;
+			memset(mem + 4096, 0, 4096);
 
 		ns = nvme_alloc_ns(dev, i, mem, mem + 4096);
 		if (ns)
-- 
1.8.5.2

